aspnet/mvc/views/razor.rst

```diff
@@ -1,8 +1,729 @@
+
+ ``<a href="mailto:Support@contoso.com">Support@contoso.com</a>``
+
+Implicit Razor expressions
+---------------------------
+
+Implicit Razor expressions start with ``@`` followed by C# code. For example:
+
+.. code-block:: none
+
+  <p>@DateTime.Now</p>
+  <p>@DateTime.IsLeapYear(2016)</p>
+
+Implicit expressions generally cannot contain spaces. For example, in the code below, one week is not subtracted from the current time:
+
+.. literalinclude:: razor/sample/Views/Home/Contact.cshtml
+  :language: html
+  :start-after: @* End of greeting *@
+  :end-before: @*Add () to get correct time.*@
+
+Which renders the following HTML:
+
+.. code-block:: none
+
+  <p>
+    Last week: 7/7/2016 4:39:52 PM - TimeSpan.FromDays(7)
+  </p>
+
+Adding parenthesis fixes the problem:
+
+.. literalinclude:: razor/sample/Views/Home/Contact.cshtml
+  :language: html
+  :start-after: @*Add () to get correct time.*@
+  :end-before: @*End of correct time*@
+
+Which renders the following HTML:
+
+.. code-block:: none
+
+  <p>
+    Last week: 6/30/2016 4:39:52 PM
+ </p>
+
+With the exception of the C# ``await`` keyword implicit expressions must not contain spaces. For example, you can intermingle spaces as long as the C# statement has a clear ending:
+
+.. code-block:: none
+
+  <p>@await DoSomething("hello", "world")</p>
+
+Explicit Razor expressions
+----------------------------
+
+Explicit Razor expressions consists of an @ symbol with balanced parenthesis. For example, to render last weeks’ time:
+
+.. code-block:: none
+
+  <p>Last week this time: @(DateTime.Now - TimeSpan.FromDays(7))</p>
+
+Any content within the @() parenthesis is evaluated and rendered to the output.
+
+Expression encoding
+-------------------
+
+Non-:dn:iface:`~Microsoft.AspNetCore.Html.IHtmlContent` content is HTML encoded. For example, the following Razor markup:
+
+.. code-block:: none
+
+  @("<span>Hello World</span>")
+
+Renders this HTML:
+
+.. code-block:: none
+
+  &lt;span&gt;Hello World&lt;/span&gt;
+
+:dn:cls:`~Microsoft.AspNetCore.Mvc.ViewFeatures.HtmlHelper` :dn:method:`~Microsoft.AspNetCore.Mvc.ViewFeatures.HtmlHelper.Raw` output is not encoded but rendered as HTML markup. :dn:cls:`~Microsoft.AspNetCore.Mvc.ViewFeatures.HtmlHelper` implements :dn:iface:`~Microsoft.AspNetCore.Html.IHtmlContent`
+
+.. warning:: Using ``HtmlHelper.Raw`` on unsanitzed user input is a security risk. User input might contain malicious JavaScript or other exploits. Sanitizing user input is difficult, avoid using ``HtmlHelper.Raw`` on user input.
+
+The following Razor markup:
+
+.. code-block:: none
+
+  @Html.Raw("<span>Hello World</span>")
+
+Renders this HTML:
+
+.. code-block:: none
+
+  <span>Hello World</span>
+
+Which the browser renders as:
+``Hello World``
+
+Razor code blocks
+------------------
+
+Razor code blocks start with ``@`` and are enclosed by ``{}``. Unlike expressions, C# code inside code blocks is not rendered.
```

Comment:
I think there needs to be something in this section that explains how the scope of code blocks relates to other C# expressions in the Razor page. Something simple like: Code blocks and expressions in a Razor page all share the same scope.